### PR TITLE
MM-10183: Adds channel scheme API endpoint.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary
Adds API endpoint to set a channel's scheme. 

This is awaiting https://github.com/mattermost/mattermost-server/pull/8615.

#### Ticket Link
[MM-10183](https://mattermost.atlassian.net/browse/MM-10183)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added [API documentation](https://github.com/mattermost/mattermost-api-reference/pull/361)
- [x] All new/modified APIs include changes to the drivers
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: api4/channel_test.go [hunk 1]
```diff
@@ -1879,3 +1879,75 @@ func TestAutocompleteChannels(t *testing.T) {
...
+
+	// Test an invalid scheme scope.
+	_, resp = th.SystemAdminClient.UpdateChannelScheme(channel.Id, teamScheme.Id)
+	fmt.Printf("resp: %+v\n", resp)
+	CheckBadRequestStatus(t, resp)
+
+	// Test that an unauthenticated user gets rejected.
+	th.SystemAdminClient.Logout()
+	_, resp = th.SystemAdminClient.UpdateChannelScheme(channel.Id, channelScheme.Id)
+	CheckUnauthorizedStatus(t, resp)
```
Comment (jespino) [inline api4/channel_test.go]: Only a suggestion if you want to do that. Here we can use table testing with fileds `license bool, loggedIn bool, channelId string, schemaId string`.

File: app/channel.go [hunk 2]
```diff
@@ -354,6 +354,24 @@ func (a *App) UpdateChannel(channel *model.Channel) (*model.Channel, *model.AppE
...
+	var err *model.AppError
+	if oldChannel, err = a.GetChannel(channel.Id); err != nil {
+		return nil, err
+	}
+
+	a.InvalidateCacheForChannel(channel)
+
+	oldChannel.SchemeId = channel.SchemeId
+
+	if result := <-a.Srv.Store.Channel().Update(oldChannel); result.Err != nil {
```
Comment (jespino) [inline app/channel.go]: Probably here makes sense to use the a.UpdateChannel directly, do exactly the same in the database and also send the websocket update of the channel (something needed for the correct behavior in the frontend).

Comment (mkraft) [reply app/channel.go]: @jespino Good idea; cleaner. Switched.

**Comment by mkraft** [comment]
Closing in favour of combined PR: https://github.com/mattermost/mattermost-server/pull/8680

**Pull request closed without merging**